sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/MetadataImpl.scala

```diff
@@ -194,6 +196,13 @@ private[kalix] class MetadataImpl(val entries: Seq[MetadataEntry]) extends Metad
 
   override def clearTime(): MetadataImpl = remove(MetadataImpl.CeTime)
 
+  override def withHttpResponseCode(code: StatusCode): MetadataImpl = {
```

Comment:
We can overload this method. One receiving `Success` and one receiving `Redirect`. As such, we don't need to add a runtime exception. 

Alternatively, we can have it as `withSuccessStatusCode` and `withRedirectStatusCode`, but I'm not convinced myself with the names. 🤷 

As a side comment, why are we planning to support redirects? Feels like unusual for a service that is mainly about APIs and less about http resources, but I might be missing some context.

Reply:
> As a side comment, why are we planning to support redirects? Feels like unusual for a service that is mainly about APIs and less about http resources, but I might be missing some context.

How else can I support my super fun KLX URL shortener? Just kidding.

In any case, since we are providing a tool for (also) building HTTP APIs, I think we need to give people that flexibility. It will be used sparsely but it has applications I think. Even for things like API deprecation cycles, it can be useful to redirect clients to newer versions.